packages/contracts/src/zksync/TokenVotingSetupZkSync.sol

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity 0.8.17;
+
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {ERC165Checker} from "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import {IERC20Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import {IVotesUpgradeable} from "@openzeppelin/contracts-upgradeable/governance/utils/IVotesUpgradeable.sol";
+
+import {IDAO} from "../core/dao/IDAO.sol";
+import {DAO} from "../core/dao/DAO.sol";
+import {PermissionLib} from "../core/permission/PermissionLib.sol";
+import {PluginSetup, IPluginSetup} from "../framework/plugin/setup/PluginSetup.sol";
+
+import {GovernanceERC20} from "../token/ERC20/governance/GovernanceERC20.sol";
+import {GovernanceWrappedERC20} from "../token/ERC20/governance/GovernanceWrappedERC20.sol";
+
+import {IGovernanceWrappedERC20} from "../token/ERC20/governance/IGovernanceWrappedERC20.sol";
+import {MajorityVotingBase} from "../plugins/governance/majority-voting/MajorityVotingBase.sol";
+import {TokenVoting} from "../plugins/governance/majority-voting/token/TokenVoting.sol";
+
+/// @title TokenVotingSetup
+/// @author Aragon Association - 2022-2023
+/// @notice The setup contract of the `TokenVoting` plugin for the ZkSync network.
+contract TokenVotingSetupZkSync is PluginSetup {
+    using Address for address;
+    using Clones for address;
+    using ERC165Checker for address;
+
+    /// @notice The address of the `TokenVoting` base contract.
+    TokenVoting private immutable tokenVotingBase;
+
+    /// @notice The token settings struct.
+    /// @param addr The token address. If this is `address(0)`, a new `GovernanceERC20` token is deployed. If not, the existing token is wrapped as an `GovernanceWrappedERC20`.
+    /// @param name The token name. This parameter is only relevant if the token address is `address(0)`.
+    /// @param symbol The token symbol. This parameter is only relevant if the token address is `address(0)`.
+    struct TokenSettings {
+        address addr;
+        string name;
+        string symbol;
+    }
+
+    /// @notice Thrown if token address is passed which is not a token.
+    /// @param token The token address
+    error TokenNotContract(address token);
+
+    /// @notice Thrown if token address is not ERC20.
+    /// @param token The token address
+    error TokenNotERC20(address token);
+
+    /// @notice Thrown if passed helpers array is of wrong length.
+    /// @param length The array length of passed helpers.
+    error WrongHelpersArrayLength(uint256 length);
+
+    /// @notice The contract constructor deploying the plugin implementation contract and receiving the governance token base contracts to clone from.
```

Comment:
I see in the code that this function no longer receives the base token contracts to deploy either a clone or a UUPS proxy, so maybe this comment could be updated 
```suggestion
    /// @notice The contract constructor deploying the plugin implementation contract.
```